src/javascript/app/pages/trade/contract.js

```diff
@@ -191,6 +193,10 @@ const Contract = (() => {
             trade_contract_forms.inout = localize('In/Out');
         }
 
+        if (trade_contract_forms.touchnotouch) {
+            trade_contract_forms.touchnotouch = localize('Touch/No Touch');
+        }
```

Comment:
I don't get what's purpose of this `if` block! the `trade_contract_forms` is initially an empty object `{}`
then on line 163 we are setting the `touchnotouch` property to `localize('Touch/No Touch')` and here we are checking if it has that property, we are setting it again one more time!
Why? if that object doesn't have that property this block is doing nothing and if the property is equal to `localize('Touch/No Touch')` we are setting the same value one more time! (this property can't have any other value in the current code base)

Reply:
'Touch/No Touch' localization in the code is different from what we get from `contract_category_display` in `contracts_for` API response, to make it consistent with the code I add this. so when the contract is `touchnotouch` we use code localize instead of getting it from API.